nemo/collections/nlp/modules/common/transformer/transformer.py

```diff
@@ -246,8 +264,18 @@ def input_example(self):
         sample = next(self.parameters())
         input_ids = torch.randint(low=0, high=2048, size=(2, 16), device=sample.device)
         encoder_mask = torch.randint(low=0, high=1, size=(2, 16), device=sample.device)
-        return tuple([input_ids, encoder_mask, self._embedding(input_ids), encoder_mask])
+        mem_size = [2, self.num_states, 15, self._hidden_size]
+        decoder_mems = torch.rand(mem_size, device=sample.device)
+        return tuple([input_ids, encoder_mask, self._embedding(input_ids), encoder_mask, decoder_mems])
 
     def _prepare_for_export(self, **kwargs):
         self._decoder.diagonal = None
+        self.return_mems = True
         super()._prepare_for_export(**kwargs)
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        if self.return_mems:
+            return {"last_hidden_states": NeuralType(('B', 'D', 'T', 'D'), ChannelType())}
```

Comment:
Shouldn't this be batch x num_layers x time x dim? Why is the dim axis present twice?

Reply:
NeMo does not have a AxisKind to represent num_layers specifically. While there is height, export considers that as a dynamic axis which might be worse to optimize. So AxisKind.Dimension looked like the best fit.